janusgraph-core/src/main/java/org/janusgraph/core/attribute/Geo.java

```diff
@@ -108,6 +108,34 @@ public boolean hasNegation() {
         public JanusGraphPredicate negate() {
             throw new UnsupportedOperationException();
         }
+    },
+
+    /**
+     * Whether one geographic region completely contains another
+     */
+    CONTAINS {
+        @Override
+        public boolean test(Object value, Object condition) {
+            Preconditions.checkArgument(condition instanceof Geoshape);
+            if (value == null) return false;
+            Preconditions.checkArgument(value instanceof Geoshape);
+            return ((Geoshape) value).contains((Geoshape) condition);
+        }
+
+        @Override
+        public String toString() {
+            return "contains";
+        }
+
+        @Override
+        public boolean hasNegation() {
+            return false;
```

Comment:
comment on why this is not supported. I think it is because when shapes intersect at their edges, part of the shapes are on the outside.